### ISSUE TITLE
Solve environment without installing.

### ISSUE THREAD

**Issue by chebee7i** [conda/conda#10379]
I did some googling and didn't see this. Apologies if the feature already exists for `conda env`. 

Due to package conflicts, it can be a very tedious process to know if it is possible to install a yml file.  The process is made worse since it can take a LONG time for the conflict report to show up (and in many cases, the conflict report is just not that insightful).

It would be nice if we could:

* solve the environment given a yml file and stop immediately, before attempting to install the env.
* ability to quickly fail if conflicts are detected (thereby giving the user an option to avoid waiting 30+ minutes for a conflict report to be generated)

The idea here is that it should be easier to create automation that let's one quickly identify if a yml file could be installed without issues.

Currently, I have do the following:
1) modify yml
2) attempt install 
3) notice a conflict 
4) Ctrl-C to get out
5) Go to step 1

In many instances, I can't even go directly to step 1 since the Ctrl-C puts my environment in a bad state. So then I have to deactivate and reactivate.  Then I can reattempt an install.

**Comment by mfansler** [comment]
#7495 would address first part.

**Comment by chebee7i** [comment]
This is still unresolved and it should not be closed.

**Comment by jaimergp** [comment]
`conda env` supports the `--dry-run` flag, which should provide what you need.

**Comment by chebee7i** [comment]
Please reread the issue description. No,  `--dry-run` does not provide what I need.  When it hits a conflict, it just churns. That's not helpful and my description makes that quite clear.

**Comment by jaimergp** [comment]
Sorry, I got it now.

TLDR: It's tricky.

There are two settings that offer some configuration options: 

```
# # unsatisfiable_hints (bool)
# #   A boolean to determine if conda should find conflicting packages in
# #   the case of a failed install.
# # 
# unsatisfiable_hints: true

# # unsatisfiable_hints_check_depth (int)
# #   An integer that specifies how many levels deep to search for
# #   unsatisfiable dependencies. If this number is 1 it will complete the
# #   unsatisfiable hints fastest (but perhaps not the most complete). The
# #   higher this number, the longer the generation of the unsat hint will
# #   take. Defaults to 3.
# # 
# unsatisfiable_hints_check_depth: 2
```

You would assume that setting the first to `false` or the second to `0` would give you something like what you need. But turns out that the unsatisfiable hints logic is the same logic used to relax excessive constraints while `conda` classic solves an environment. If something is reported as a conflict in the early iterations and the version constraint can be relaxed (because that pin comes from secondary sources, e.g. it's an installed dependency of a package you did request), conda will remove the version constraint and try again.

I investigated _why_ in this PR, but my research back then didn't go anywhere: https://github.com/conda/conda/pull/10867. A fix for this would involve adding one more setting to disable the "ok we tried everything and didn't work, let's tell the user what we think it's wrong" part.

I'd say your best bet right now is changing solvers and giving `conda-libmamba-solver` a try if you can. It's not only faster, but conflicts are informative.
